databuilder/query_engines/sqlite.py

```diff
@@ -68,7 +68,11 @@ def select_patient_id_for_population(self, population_expression):
             # If there's only one table then use the IDs from that
             return sqlalchemy.select(tables[0].c.patient_id)
         else:
-            assert False, "No tables referenced in population expression"
+            # Gracefully handle the degenerate case where the population expression
+            # doesn't reference any tables at all. Our validation rules ensure that such
+            # an expression will never evaluate True so the population will always be
+            # empty. But we can at least return an empty result, rather than blowing up.
+            return sqlalchemy.select(sqlalchemy.literal(None).label("patient_id"))
```

Comment:
What's the reason for handling this? I think it means that we support a population of `Value(False)`, but not one of `Value(True)`, which seems odd.

Reply:
Yes, you're right that we support `Value(False)` but not `Value(True)` and that is a bit odd. It would be reasonable, I think, to reject any population expression which is always False regardless of its inputs. _However_ detecting that class of expression requires a whole load more code similar to, but not the same as, the `evaluate` functions in the current population_validation module. Whereas gracefully handling it like this requires one line.

The alternative is not to handle it and blow up, but I wanted to preempt the smug undertones in Hypothesis's voice — "well, you did _say_ this was a valid population ..."

Reply:
Yes, I think you're right that we would need to add the validation. I suppose that's the ideal, most-friendly-possible outcome (on the assumption that any population that always evaluates to `False` must be an error which should be caught as early as possible). But I don't have any reservations about this going in as it is now.